pkg/scalers/metrics_api_scaler.go

```diff
@@ -63,6 +124,61 @@ func metricsAPIMetadata(resolvedEnv, metadata, authParams map[string]string) (*m
 		return nil, fmt.Errorf("no valueLocation given in metadata")
 	}
 
+	// no authMode specified
+	if _, ok := authParams["authMode"]; !ok {
+		return &meta, nil
+	}
+
+	val, _ := authParams["authMode"]
+	authType := authenticationType(strings.TrimSpace(val))
+	if authType == apiKeyAuth {
+		if len(authParams["apiKey"]) == 0 {
```

Comment:
```suggestion
		if len(authParams["apiKey"]) == 0 {
```
I think this may result in an error as the key may not exists

Reply:
I am sensing your concern is with len function. But this will not fail. authParams is a map of string. string can never be nil in go, that's why len will never fail.

Reply:
Oh, interesting! I didn't know. No objections then 🚀